debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -396,6 +396,14 @@ public ReplicationStream startStreaming(Lsn offset, WalPositionLocator walPositi
         initConnection();
 
         connect();
+
+        if (connectorConfig.isYSQLMajorUpgrade()) {
```

Comment:
Can we also check the places where we retry in case of failures

Reply:
In case of a failure, the PostgresConnectorTask will encounter a retriable exception and will retry. In the next attempt when it tries to stream again it will encounter this code block again and will set the GUC based on the config value.